File: leecode/70/70.cpp
#include <iostream>

class Solution
{
public:
  int climbStairs(int n)
  {
    if (n <= 2)
      return n;
    int dp[46];
    dp[0] = 1;
    dp[1] = 2;
    for (int i = 2; i < n; i++)
    {
      dp[i] = dp[i - 1] + dp[i - 2];
    }
    return dp[n - 1];
  }
};
int main()
{
  Solution solution;
  int step = solution.climbStairs(3);
  std::cout << step << std::endl;
  return 0;
}

File: leecode/70/70.py
class Solution:
    def climbStairs(self, n: int) -> int:
      if n == 1:
        return 1
      elif n == 2:
        return 2
      dp = [0]*(n+1);
      dp[0], dp[1], dp[2] = 0, 1, 2
      for i in range(3,n+1):
        dp[i] = dp[i-1] + dp[i-2]
      return dp[n]
print(Solution().climbStairs(10))